src/haddock/modules/__init__.py

```diff
@@ -53,6 +54,30 @@
     ".cfg": read_config,
     }
 
+_step_folder_regex = tuple(
+    r"[0-9]+_" + mod_name
+    for mod_name in modules_category.keys()
+    )
+step_folder_regex = "(" + "|".join(_step_folder_regex) + ")"
```

Comment:
I do not think you need the group here (braces) as the group is not used anywhere and does not change the behaviour of the pipe `|`.

Reply:
Another thing, here on line 58 in the regex instead of `[0-9]` you could use `\d` as this does the same.